### DISCUSSION TITLE
Help with using Paraview for visualizing

### DISCUSSION THREAD

**Discussion by ColinMG2** [idaholab/moose#28784] [Q&A Tools] [answered]
I have been trying to use paraview through conda in the Ubuntu terminal to visualize a _in.e file. Here is my error: qt.qpa.xcb: could not connect to display 10.255.255.254:0.0
qt.qpa.plugin: From 6.5.0, xcb-cursor0 or libxcb-cursor0 is needed to load the Qt xcb platform plugin.
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: vkkhrdisplay, vnc, xcb, offscreen, minimal, linuxfb, eglfs, minimalegl.

Does anyone know how to resolve this issue?

**Comment by GiudGiud** [comment]
Hello

For now you should simply install paraview on the windows side and use that.
I recall @milljm had some good first steps to add missing packages on the Ubuntu side. I dont recall if that was enough

I think we recommended installing paraview from the paraview website on the Ubuntu side. This takes care of some missing dependencies in conda

Guillaume

**Reply by milljm** [reply]
```pre
10.255.255.254:0.0
```
This makes me think you are operating within a WSL instance? If you are, there are sometimes additional steps one needs to take to allow Linux GUI apps to connect to the Windows GUI.

On top of the things listed on our [WSL Instructions](https://mooseframework.inl.gov/getting_started/installation/windows.html), you may find additional tricks to try from this long thread: https://github.com/idaholab/moose/discussions/22718#discussioncomment-4276193.

**Reply by ColinMG2** [reply]
Yes I am using WSL. I have been trying to find the additional steps to allow Linux GUI apps to connect to the Windows GUI but I can't seem to find them.

**Reply by milljm** [reply]
I would start here: https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps

And see if you can get any of the examples they describe working (like GIMP).

**Reply by ColinMG2** [reply]
I was able to get Tilix (a virtual Linux terminal) to work. I am actually able to run moose on there but I still run into the same error message when I try to run the "paraview" command on a _in.e or _out.e file.

**Reply by ColinMG2** [reply, marked as answer]
I was able to get it to work by running the following: 
sudo apt install xterm
export DISPLAY=:1.0
xterm

The output was as follows:
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
xterm is already the newest version (372-1ubuntu1).
0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.
xterm: Xt error: Can't open display: :1.0

By running export DISPLAY=:0 this fixes the error I got in the output message. I can now run paraview through a conda environment and it will open a Linux GUI for Paraview.

**Reply by milljm** [reply]
Interesting, usually the IP address is required. I'll mark your post as the answer. Let me know if this does not work for you!

**Reply by ColinMG2** [reply]
This link here helped me out with this problem. https://forum.qt.io/topic/132855/qt-qpa-xcb-could-not-connect-to-display-1-0-qt-qpa-plugin-could-not-load-the-qt-platform-plugin-xcb-in-even-though-it-was-found
